packages/kit/src/core/postbuild/prerender.js

```diff
@@ -360,9 +398,19 @@ async function prerender({ hash, out, manifest_path, metadata, verbose, env }) {
 		const type = headers['content-type'];
 		const is_html = response_type === REDIRECT || type === 'text/html';
 
-		const file = output_filename(decoded, is_html);
+		const file = output_filename(decoded, is_html, type);
 		const dest = `${config.outDir}/output/prerendered/${category}/${file}`;
 
+		// Warn about prerendering non-HTML root endpoints
+		if (decoded === config.paths.base + '/' && !is_html && category === 'pages') {
+			log.warn(
+				`Prerendering root +server.js endpoint with content-type '${type}'. ` +
+					`Note: Most static hosts expect 'index.html' at the root. ` +
```

Comment:
Not sure if we need this warning because the user might not be deploying to a static host. e.g., I can configure to prerender the root endpoint but I don't have to prerender all other pages (unless the user is using the static adapter which is not guaranteed here).